### ISSUE TITLE
Export manually lay outed diagrams with Structurizr Lite

### ISSUE THREAD

**Issue by mh182** [docToolchain/docToolchain#1057]
Based on #1016 provide the means to export diagrams which were manually lay outed following the workflow as described in the [documentation of Structurizr Lite](https://structurizr.com/share/76352/documentation#workflow).

**Describe the solution you'd like**
Enhance the existing task `exportStructurizr` so that it processes the layout file in case the workspace is provided as `JSON` file. We have to make sure the `JSON` file is up-to-date in case the `DSL` file is newer.

**Non-Goals**
* Integration of `Structurizr Lite` in `docToolchain`
* Processing of multiple workspace projects.

**Additional context**

**Comment by simonbrowndotje** [comment]
From #1058:

> We want to support both: generating PlantUML from Structurizr DSL and from manually lay-outed diagrams with Structurizr (which leads to a JSON file).

I just want to clarify something ... you *can't* use PlantUML to export diagrams with a manual layout. The only diagram renderer that supports manual layout is the renderer you can find at https://github.com/structurizr/ui ... this is a JavaScript/JointJS renderer designed to run in a web browser. So the diagrams are rendered client-side, i.e. in a web browser, and it's this UI that allows people to move boxes and arrows.

If you want to export these manual layout diagrams, you will need to use the built-in SVG/PNG export feature provided by Structurizr Lite/on-premises/cloud -> https://structurizr.com/help/exporting-diagrams (i.e. there's a button on the UI that you will need to click).

This can be scripted, but you will need to use something like headless Chrome and Puppeteer. I have an example script that will do this (https://github.com/structurizr/puppeteer#exporting-diagrams), but you will need to add node, npm, chromium, and puppeteer as additional dependencies to docToolChain.

**Comment by mh182** [comment]
> 

Thanks for the clarification.

This was a mistake from my side. I was misled by the Structurizr CLI which is able to process JSON and DSL files. I thought (but never tested) that the `export` command using a JSON file would render manual lay-outed diagrams.

This really is a stopper for the design I had in mind. 

I really have to think about how we integrate the information of Structurizr models:
- using `generateHTML` we could use the JavaScript render (read only) so people could navigate the model interactively (which is a plus IMHO)
- but for `generatePDF` we would need to export the diagrams from the model to SVG/PNG. The proposed headless Chrome+Puppeteer seems quite complicate to achieve this.

**Comment by simonbrowndotje** [comment]
> Thanks for the clarification.

You're welcome. [Site Generatr](https://github.com/avisi-cloud/structurizr-site-generatr) hyperlinks the PlantUML exports together ([example](https://avisi-cloud.github.io/structurizr-site-generatr/main/internet-banking-system/context/)), so there are certainly other methods for building a more interactive version of the diagrams.

**Comment by mh182** [comment]
> > Thanks for the clarification.
> 
> You're welcome. [Site Generatr](https://github.com/avisi-cloud/structurizr-site-generatr) hyperlinks the PlantUML exports together ([example](https://avisi-cloud.github.io/structurizr-site-generatr/main/internet-banking-system/context/)), so there are certainly other methods for building a more interactive version of the diagrams.

Following our discussion above, I assume that the diagrams generated by `Site Generatr` use auto-layout, correct?

Seems that docToolchain and `Site Generatr` try to solve the same problem.

**Comment by simonbrowndotje** [comment]
> I assume that the diagrams generated by Site Generatr use auto-layout, correct?

Yes, it uses an extended version of the C4PlantUMLExporter that overrides element URLs -> https://github.com/avisi-cloud/structurizr-site-generatr/blob/main/src/main/kotlin/nl/avisi/structurizr/site/generatr/site/C4PlantUmlExporterWithElementLinks.kt

> Seems that docToolchain and Site Generatr try to solve the same problem.

Yes.
